packages/main/src/plugin/index.ts

```diff
@@ -1037,9 +1041,20 @@ export class PluginSystem {
         imageName: string,
         callbackId: number,
       ): Promise<void> => {
-        return containerProviderRegistry.pullImage(providerContainerConnectionInfo, imageName, (event: PullEvent) => {
-          this.getWebContentsSender().send('container-provider-registry:pullImage-onData', callbackId, event);
-        });
+        return progressImpl.withProgress<void>(
+          { location: ProgressLocation.TASK_WIDGET, title: `Pulling ${imageName}.` },
+          async progress => {
+            await containerProviderRegistry.pullImage(
+              providerContainerConnectionInfo,
+              imageName,
+              (event: PullEvent) => {
+                this.getWebContentsSender().send('container-provider-registry:pullImage-onData', callbackId, event);
+              },
+            );
+            progress.report({ increment: -1 });
+            notificationImpl.showNotification({ body: `${imageName} pulled.` });
```

Comment:
Notifications does not appear for me.

Reply:
Correction, on macOS it does not appear right away. One would have to go and click on date/time part of the tray in upper right corner. On fedora 38 it seems not working at all.